core/ringbuffer.go

```diff
@@ -121,3 +121,23 @@ type policy int32
 func (op policy) Policy() policy {
 	return op
 }
+
+// The result of a Ringbuffer.ReadMany operation
```

Comment:
In go the convention for docs is to start with what you are documenting, for example here it should be `ReadResultSet is the result of a .... ` and in methods it is the same `ReadCount returns the... ` etc